tests/kafkatest/services/kafka/kafka.py

```diff
@@ -79,7 +79,7 @@ def __init__(self, context, num_nodes, zk, security_protocol=SecurityConfig.PLAI
         :type topics: dict
```

Comment:
Not really related to this pr - I was looking at the surrounding code and noticed that server_prop_overides is defaulted to `[]`. Its probably fine since its never being read, but you almost never want that since every instance will share the same global list. Might want to fix that while you're here.

Reply:
@rodesai what do you mean by the "global list"?

Reply:
When the interpreter processes the function def it will initialize one list that it passes to every instance of KafkaService thats constructed without specifying its own server_prop_overrides. Here's an illustrative example:
```code
>>> class A(object):
...     def __init__(self, a, b=[]):
...         b.append(a)
...         self.b = b
...
>>> print A(1).b
[1]
>>> print A(2).b
[1, 2]
>>> print A(3).b
[1, 2, 3]
```

Reply:
Got it, thanks!

Reply:
This still has the same problem, though its more explicit. Not sure if that was the intent. The standard approach is something like this
```{code}
def __init__(self, ..., server_prop_overrides=None, ...):
    if server_prop_overrides is None:
        server_prop_overrides = []
```

Reply:
Ah got it. I will try this in the code:

```
if server_prop_overides is None:
            self.server_prop_overides = []
else:
            self.server_prop_overides = server_prop_overides
```